notification-portlet-webapp/src/main/webapp/scripts/modal-notice.js

```diff
@@ -0,0 +1,162 @@
+        actions:         '.np-actions',
+        actionTemplate:  '.np-action-template',
+        closeButton:     '.np-close'
+      }
+    };
+
+    // First 'prime-the-pump' with an ActionURL
+    var initNotices = function($, settings, callback) {
+      $.ajax({
+        type:    'POST',
+        url:     settings.invokeNotificationServiceUrl,
+        async:   false,
+        success: function() {
+          fetchNotices($, settings, callback);
+        }
+      });
+    }
+
+    // Then fetch the notifications with a ResourceURL
+    var fetchNotices = function($, settings, callback) {
+      $.ajax({
+        url:      settings.getNotificationsUrl,
+        type:     'POST',
+        dataType: 'json',
+        success:  function (data) {
+          feed = data.feed;
+          callback(feed);
+        },
+        error:    function () {
+          container.html(" ").text("AJAX failed. ~ THE END ~");
+        }
+      });
+    }
+
+    upmodal_notice.launch = function ($, container, options) {
+
+      var settings = $.extend({}, defaults, options);
+
+      var handleAction = function() {
+        var actionButton = $(this);
+        var formElement = container.find(settings.selectors.form);
+        $.post(actionButton.attr('data-url'), formElement.serialize(), function(data, textStatus, jqXHR) {
+          if (console && console.log) {
+            console.log('handleAction received status: ' + textStatus);
+          }
+        });
+        // Any action closes the modal window
+        container.modal('hide');
+      }
+
+      var resetDialog = function() {
+        container.off('hidden.bs.modal'); // Clear previous event handler(s)
+        container.find(settings.selectors.closeButton).hide();
+        container.find(settings.selectors.actions).find('.np-action').remove();
+      }
+
+      var drawActions = function(alert) {
+        var actionsContainer = container.find(settings.selectors.actions);
+        var availableActions = alert.availableActions;
+        var actionTemplate = actionsContainer.find(settings.selectors.actionTemplate);
+        if (availableActions && availableActions.length > 0) {
+          // Draw actions if we have them...
+          for (var i=0; i < availableActions.length; i++) {
+            var action = availableActions[i];
+            var actionUrl = settings.invokeActionUrlTemplate
+                            .replace('NOTIFICATIONID', alert.id)
+                            .replace('ACTIONID', action.id);
+            var actionElement = actionTemplate.clone();
+            actionElement.removeClass('np-action-template');
+            actionElement.toggleClass('hidden');
+            actionElement.addClass('np-action');
+            if (i == 0) {
+              // Only the first button is btn-primary
+              actionElement.find('a').addClass('btn-primary');
+            }
+            actionElement.find('a').attr('data-url', actionUrl).html(action.label).click(handleAction);
```

Comment:
:art: Maybe use the more explicit `.on('click', handleAction);`?
`.click` is a risky alias in jQuery because `.click()` is an alias for `.trigger('click')`, while `.click(cb)` is an alias for `.on('click', cb);`
Ref: https://api.jquery.com/click/#click-handler